src/mscorlib/shared/System/HashCode.cs

```diff
@@ -323,33 +323,30 @@ public void Add<T>(T value, IEqualityComparer<T> comparer)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void Add(int value)
         {
-            // xxHash works as follows:
-            // 0. Initialize immediately. We can't do this in a struct (no default
-            //    ctor).
+            // The original xxHash works as follows:
+            // 0. Initialize immediately. We can't do this in a struct (no
+            //    default ctor).
             // 1. Accumulate blocks of length 16 (4 uints) into 4 accumulators.
-            // 2. Accumulate remaining blocks of length 4 (1 uint) into the hash.
+            // 2. Accumulate remaining blocks of length 4 (1 uint) into the
+            //    hash.
             // 3. Accumulate remaining blocks of length 1 into the hash.
 
-            // There is no need for *3 as this type only accepts ints. _queue1,
-            // _queue2 and _queue3 are basically a buffer so that when ToHashCode is
-            // called we can execute *2 correctly. That's what first three case
-            // statements do.
+            // There is no need for #3 as this type only accepts ints. _queue1,
+            // _queue2 and _queue3 are basically a buffer so that when
+            // ToHashCode is called we can execute #2 correctly.
 
-            // We need to initialize the xxHash32 state (_v1 -> _v4) lazily (see *0)
-            // and the last place that can be done if you look at the original code
-            // is just before the first block of 16 bytes is mixed in. The xxHash32
-            // state is never used for streams containing fewer than 16 bytes.
+            // We need to initialize the xxHash32 state (_v1 to _v4) lazily (see
+            // #0) nd the last place that can be done if you look at the
+            // original code is just before the first block of 16 bytes is mixed
+            // in. The xxHash32 state is never used for streams containing fewer
+            // than 16 bytes.
 
-            // A bloom filter is used to determine whether the default case statement
-            // has even been executed. To do that we check if the length is smaller
-            // than 4 (_length ^ position will be non-zero if this is the case). The
-            // case statement is for values larger than 2, so the check only succeeds
-            // on exactly 3.
-
-            // To see what's really going on here, have a look at the Combine methods.
+            // To see what's really going on here, have a look at the Combine
+            // methods.
             
             var val = (uint)value;
-            uint position = _length % 4;
+            uint previousLength = _length++;
```

Comment:
Storing this locally shaves off many bytes in the final code.

Reply:
If that's important, then please add a comment about it.